LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaMessage/LoRaPayloadJoinRequest.cs

```diff
@@ -57,10 +53,10 @@ public LoRaPayloadJoinRequest(byte[] inputMessage)
             // get the DevEUI
             DevEUI = new Memory<byte>(inputMessage, 9, 8);
             // get the DevNonce
-            DevNonce = new Memory<byte>(inputMessage, 17, 2);
+            DevNonce = DevNonce.Read(inputMessage[17..19].AsSpan());
```

Comment:
Slicing the array directly with `[17..19]` creates a _copy_ of the array slice. What you want here is a span over the source array:

```suggestion
            DevNonce = DevNonce.Read(inputMessage.AsSpan(17));
```